frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java

```diff
@@ -1128,7 +1128,9 @@ public void updateRelease(ActionRequest request, ActionResponse response) throws
     }
 
     private void prepareRequestForReleaseEditAfterDuplicateError(ActionRequest request, Release release) throws TException {
-        fillVendor(release);
+        if(!isNullOrEmpty(release.getVendorId())) {
```

Comment:
I am sorry for having to say something in every ticket. But I was just asking myself, if the fillVendor method itself is not badly implemented. Because this one tries to check its precondition but might just do it wrong. Because it only checks if the vendorid has been set and not that it is set different from null or empty.
Even though I am not a 100% sure if the vendor should be removable once it has been set (for which setting it to null or empty explicitly would be necessary), but this code paths would break in this case as well, wouldn't it?
So maybe better fix fillVendor in case that anyone else might want to use this method in this class.

Reply:
Thanks @imaykay for the suggestion, As of now there is only one execution flow for fillVendor() so my idea was to not enter into the flow if vendorId is missing which is happening for new release .Yes you are right if tomorrow someone else tries to use the method it may break.I will make changes inside fillVendor() method before thrift call (thriftClients.makeVendorClient()) to save the flow from crashing.

Reply:
I have added a check inside fillVendor() method. I have not removed any of the existing check as I am not sure about the implication of removing them.

Reply:
Hi @nutanv1contr and again sorry for being picky! But why don't we go with a combination of both if clauses like
`if(release.isSetVendorId() && !isNullOrEmpty(release.getVendorId())) {`
as the first line in the method? I think this would be the optimal solution because
* we do only create a Thrift client when we really will use it
* if the vendorid has not been set it cannot be null or empty, so it fails faster
* we still keep the existing check

But these are so minor improvements that I would also approve your change if you don't have the time to do another adjustment.

Reply:
Thanks for the suggestion.I have pushed the suggested changes.

Reply:
Thanks a lot @nutanv1contr